influxql/engine.go

```diff
@@ -669,54 +669,66 @@ func ReduceMean(key Key, values []interface{}, e *Emitter) {
 
 // MapMin collects the values to pass to the reducer
 func MapMin(itr Iterator, e *Emitter, tmin int64) {
-	var values []float64
+	var min *float64
 
 	for k, v := itr.Next(); k != 0; k, v = itr.Next() {
-		values = append(values, v.(float64))
+		if min == nil {
+			m := v.(float64)
+			min = &m
+		}
+		m := math.Min(*min, v.(float64))
+		min = &m
+	}
+	if min != nil {
+		e.Emit(Key{tmin, itr.Tags()}, *min)
 	}
-	e.Emit(Key{tmin, itr.Tags()}, values)
 }
 
 // ReduceMin computes the min of value.
 func ReduceMin(key Key, values []interface{}, e *Emitter) {
 	var min *float64
+
 	for _, value := range values {
-		vals := value.([]float64)
-		for _, v := range vals {
-			// Initialize min
-			if min == nil {
-				min = &v
-			}
-			m := math.Min(*min, v)
-			min = &m
+		v := value.(float64)
+		// Initialize min
+		if min == nil {
+			min = &v
 		}
+		m := math.Min(*min, v)
+		min = &m
 	}
 	e.Emit(key, min)
```

Comment:
That's a good question, and I don't know the answer to that. I have a test that tests that, and it passes, but I haven't looked deep enough to know why it works yet.  Perhaps @pauldix has more insight as to if I'm being to defensive or not enough on this call.